src/util/mapbox.js

```diff
@@ -119,3 +125,72 @@ function formatUrl(obj: UrlObject): string {
     const params = obj.params.length ? `?${obj.params.join('&')}` : '';
     return `${obj.protocol}://${obj.authority}${obj.path}${params}`;
 }
+
+const STORAGE_TOKEN = 'mapbox.userTurnstileData';
+const localStorageAvailable = storageAvailable('localStorage');
+
+export const postTurnstileEvent = function(tileUrls: Array<string>) {
+    //Enabled only when Mapbox Access Token is set and a source uses
+    // mapbox tiles.
+    if (!config.ACCESS_TOKEN ||
+        !tileUrls ||
+        !tileUrls.some((url) => { return /mapbox.c(n)|(om)/i.test(url); })) {
```

Comment:
`parseUrl` throws, and I'd like to avoid having a `try..catch`. Even with the `authority` parsed out of the url, it will need a check for `mapbox.com` and `mapbox.cn`.